### DISCUSSION TITLE
Why argocd-application-controller consumes very much memory and cpu?

### DISCUSSION THREAD

**Discussion by AllenZMC** [argoproj/argo-cd#10262] [Q&A] [answered]
Help！Why argocd-application-controller consumes very much memory and cpu? How to optimize?
![企业微信截图_2cf97150-f019-4283-bd49-45d1ef770f1e](https://user-images.githubusercontent.com/49428979/183808312-96e0366c-a063-4432-81fc-00c411fc3226.png)

**Comment by jgwest** [comment, marked as answer]
Hi @AllenZMC, in short, Argo CD needs to keep track of many different K8s cluster resources (both their live state and expected state), and this occupies a significant amount of memory, depending on factors like how many clusters are being watched and how many resources are being managed. Some tunables are here: https://argo-cd.readthedocs.io/en/stable/operator-manual/high_availability/#argocd-application-controller

See this previous question where a few of us went into some detail on why the application-controller is using as much memory as it does: https://github.com/argoproj/argo-cd/discussions/6964#discussioncomment-1164100

**Reply by AllenZMC** [reply]
> Hi @AllenZMC, in short, Argo CD needs to keep track of many different K8s cluster resources (both their live state and expected state), and this occupies a significant amount of memory, depending on factors like how many clusters are being watched and how many resources are being managed. Some tunables are here: https://argo-cd.readthedocs.io/en/stable/operator-manual/high_availability/#argocd-application-controller
> 
> 
> 
> See this previous question where a few of us went into some detail on why the application-controller is using as much memory as it does: https://github.com/argoproj/argo-cd/discussions/6964#discussioncomment-1164100

Thanks！
